### PR TITLE
[core] Introduce BlockCache for SortLookupStoreReader

### LINKED ISSUE #4096: [Feature] Add block cache for sorted lookup file store
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Motivation

Subtask of #3827 , add block cache for sorted file store

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Aitozi**
### Purpose

Linked issue: close #4096

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/io/CompressedPageFileInput.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public RandomAccessFile file() {
 
     @Override
     public long uncompressBytes() {
```
Comment (wwj6591812) [inline paimon-common/src/main/java/org/apache/paimon/io/CompressedPageFileInput.java:69]: method name uncompressedBytes

File: paimon-common/src/main/java/org/apache/paimon/io/cache/CacheManager.java [hunk 2]
```diff
@@ -76,4 +76,26 @@ public void invalidPage(CacheKey key) {
     public int fileReadCount() {
         return fileReadCount;
     }
+
+    /** The container for the segment. */
+    public static class SegmentContainer {
+
+        private final MemorySegment segment;
+
+        private int accessCount;
```
Comment (wwj6591812) [inline paimon-common/src/main/java/org/apache/paimon/io/cache/CacheManager.java:85]: need AtomicInteger?

File: paimon-common/src/main/java/org/apache/paimon/lookup/sort/SortLookupStoreReader.java [hunk 3]
```diff
@@ -127,64 +121,61 @@ private BlockReader openBlock(MemorySlice blockEntry) throws IOException {
...
+                                BlockDecompressor decompressor =
+                                        compressionFactory.getDecompressor();
+                                int uncompressedLength =
+                                        decompressor.decompress(
+                                                block.getHeapMemory(),
+                                                compressedInput.position(),
+                                                compressedInput.available(),
+                                                uncompressed,
+                                                0);
+                                checkArgument(uncompressedLength == uncompressed.length);
```
Comment (wwj6591812) [inline paimon-common/src/main/java/org/apache/paimon/lookup/sort/SortLookupStoreReader.java:166]: may checkState better?

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**